src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +66,12 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
+                        && ((p.getSex().equals(Person.Sex.MAN)
+                        && p.getAge() <= maleToAge)
+                        || (p.getSex().equals(Person.Sex.WOMAN) && p.getAge() <= femaleToAge)))
```

Comment:
if we have large number of checks it might be suitable to create separate predicate and place those checks there
```suggestion
                .filter(workablePredicate)
```

Reply:
```
Predicate<Person> workablePredicate = person -> { checks here }
```
you can write it just above return statement